WriteVHDLSyntax.py

```diff
@@ -178,14 +174,14 @@ def writeTBMemoryReadInstance(mtypeB, memDict, bxbitwidth, is_initial, is_binned
 
         mem = memMod.inst
 
-        if split and "in" not in mem:
+        if split == 2 and "in" not in mem:
             continue
     
         if "DL" in mtypeB and "AS" not in mtypeB: # Special case for DTC links that reads from FIFOs
             string_mem += "    read" + mem + " : entity work.FileReaderFIFO\n"
             string_mem += "  generic map (\n"
             memtmp = mem.replace("twoS","2S")
-            string_mem += "      FILE_NAME".ljust(str_len) + "=> FILE_IN_DL&\""+ memtmp + "\"&inputFileNameEnding,\n"
+            string_mem += "      FILE_NAME".ljust(str_len) + "=> FILE_IN_DL_39&\""+ memtmp + "\"&inputFileNameEnding,\n"
```

Comment:
@jasonfan393 We manage to avoid the hard-coded `39` when the `FILE_IN_DL_39` constant is defined here:
https://github.com/cms-L1TK/project_generation_scripts/blob/04753f7c6a49a1781b83221e0d5b302916b27d9c/WriteVHDLSyntax.py#L980

Can we do something similar here by using `mtypeB`?
```suggestion
            string_mem += "      FILE_NAME".ljust(str_len) + "=> FILE_IN_"+mtypeB+"&\""+ memtmp + "\"&inputFileNameEnding,\n"
```